lib/characteristic.dart

```diff
@@ -170,21 +168,22 @@ class Characteristic extends InternalCharacteristic {
 ///
 /// This type is created to support chaining of operations on the characteristic
 /// when it was first read from [Peripheral] or [Service].
-class CharacteristicWithValue extends Characteristic with WithValue {
+class CharacteristicWithValue extends Characteristic {
+  Uint8List value;
```

Comment:
I didn't get the point of the mixin here since it's not used anywhere else. 
Do you plan to use polymorphism over the type WithValue ever? I couldn't imagine any use case like for example if there was a function with the signature: `void foo(WithValue value)` I'd understand, one may pass in anything that conforms to WithValue thus has a `value` filed.

Otherwise, to me, it was cleaner and more concise to just have the field there, as jumping across files all the time takes a lot of time, for imho no benefit.
But again I might be missing some clue here.

Reply:
Only benefit was that it was used for Characteristic and Descriptor. Not much of a value, I agree.